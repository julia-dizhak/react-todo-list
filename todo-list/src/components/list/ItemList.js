import React from 'react';

export default function ItemList({items}) {
    // return items.length ? (
    //     <ul>{items.map(i => <li key={i}>{i}</li>)}</ul>
    // ) : (
    //     <span>'no items'</span>
    // )

    return (
        // <ul>{items.map(i => <li key={i}>{i}</li>)}</ul>
        <ul>test</ul>
    ) 
}

