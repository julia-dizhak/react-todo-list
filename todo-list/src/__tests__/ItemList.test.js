/**
 * @jest-environment node
*/
 
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ItemList from './../components/list/ItemList';

describe('ItemList', () => {
    it('should render component correctly', () => {
        const container = renderer.create(<ItemList />).toJSON()
        expect(container).toMatchSnapshot();
    });

    // describe('case when when no items are given', () => {
    //     // const container = document.createElement('div');
    //     // ReactDOM.render(<ItemList items={[]} />, container);

    //     const container = renderer.create(<ItemList items={[]} />).toJSON()
    //     expect(container).toMatchSnapshot();
    //     // shallow(<ItemList items={[]} />);

    //     //expect(container.innerHTML).toBe('no  items');
    //     //expect(container.textContent).toMatch('no items');
    // });

    // describe('case with the given items', () => {
    //     const container = document.createElement('div');
    //     ReactDOM.render(<ItemList items={['string1, string2']} />, container);
    //     expect(container.textContent).toMatch('string1');
    //     expect(container.textContent).toMatch('string2');
    // });
});
