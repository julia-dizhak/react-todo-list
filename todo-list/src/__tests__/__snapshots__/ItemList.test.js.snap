// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList render "no items" when no items are given 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList render component 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList render correctly component 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList renders "no items" when no items are given 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render component 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render component correctly 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render correctly component 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render correctly componentrenders "no items" when no items are given 1`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render correctly componentrenders "no items" when no items are given 2`] = `
<span>
  'no items'
</span>
`;

exports[`ItemList should render correctly componentrenders 1`] = `
<span>
  'no items'
</span>
`;
